spark_log_parser/loaders/json.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import orjson
+from typing import Generic, TypeVar, Iterator
+
+from aiodataloader import DataLoader
+
+from spark_log_parser.loaders.https import HTTPFileLinesDataLoader, HTTPFileBlobDataLoader
+from spark_log_parser.loaders.local_file import LocalFileBlobDataLoader, LocalFileLinesDataLoader
+from spark_log_parser.loaders.s3 import S3FileBlobDataLoader, S3FileLinesDataLoader
+
+RawJSONBlobDataLoader = TypeVar("BlobDataLoader", LocalFileBlobDataLoader, S3FileBlobDataLoader, HTTPFileBlobDataLoader)
+
+logger = logging.getLogger("JSONLoaders")
+
+
+class JSONBlobDataLoader(DataLoader, Generic[RawJSONBlobDataLoader]):
```

Comment:
Is this generic doing anything for us?

Reply:
Yeah, it might not be. I guess it's less useful in the Python world (instead of the Java world) where I could just replace this with a Union type and just use it in the signature of `__init__`

Reply:
I think it's still useful in Python. For example, we use this model in a couple places: https://docs.pydantic.dev/usage/models/#generic-models. In that case the generic typing tells us what type the `data` is. In this case, I don't think it tells us much of a JSONBlobDataLoader instance's interface - unless `blob_data_loader` is meant to be accessed from outside the class?

Reply:
Oh yeah, that's not to say that they aren't useful in Python - it just seems like for some of the use cases where one might use a Generic in a language like Java, someone could just use a Union type in Python.

In this case, the intent of the `RawJSONBlobDataLoader` is to indicate "where" the data is coming from (i.e. HTTP/S3/`file://`), but it's true that that inner loader is not intended to be accessed externally, and that type is not really part of the API of this class.

So I guess, kind of summing those thoughts up - in Python, perhaps Generics are better used when that type is more a part of the External API. Otherwise, a Union will likely suffice if it's just to specify "these are the valid types for this argument".